test/util.js

```diff
@@ -669,10 +669,27 @@ describe('common/util', function() {
     });
 
     it('should create an authClient', function(done) {
-      var config = {};
+      var config = {
+        test: true,
+      };
+
+      googleAutoAuthOverride = function(config_) {
+        assert.deepStrictEqual(config_, config);
+        setImmediate(done);
+        return authClient;
+      };
+
+      util.makeAuthenticatedRequestFactory(config);
+    });
+
+    it('should not pass projectId token to google-auto-auth', function(done) {
+      var config = {
+        projectId: '{{projectId}}',
+      };
 
       googleAutoAuthOverride = function(config_) {
-        assert.strictEqual(config_, config);
+        assert.strictEqual(config_.projectId, undefined);
+        assert.strictEqual(config.projectId, '{{projectId}}');
```

Comment:
this check is not needed: we're testing that `{{projectId}}` is not passed down, not that it was originally assigned correctly.

Reply:
or is it here to verify that the caller's object is kept unchanged? if so, it should be in a different small unit test.

Reply:
This is a confirmation that we didn't delete it from the real config object.

Reply:
Yeah, then the name `'should not pass projectId token to google-auto-auth'` is misleading.  (but it's a minor thing of course)

Reply:
Split 'em!